src/timeline/InputBar.cpp

```diff
@@ -249,6 +250,54 @@ InputBar::updateAtRoom(const QString &t)
         emit containsAtRoomChanged();
     }
+
+    // check for invalid commands
+    auto commandName = getCommandAndArgs().first;
+    bool hasInvalidCommand{};
+    if (!commandName.isNull() && '/' + commandName != text()) {
+        static const QStringList validCommands{QStringLiteral("me"),
+                                               QStringLiteral("react"),
+                                               QStringLiteral("join"),
+                                               QStringLiteral("knock"),
+                                               QStringLiteral("part"),
+                                               QStringLiteral("leave"),
+                                               QStringLiteral("invite"),
+                                               QStringLiteral("kick"),
+                                               QStringLiteral("ban"),
+                                               QStringLiteral("unban"),
+                                               QStringLiteral("redact"),
+                                               QStringLiteral("roomnick"),
+                                               QStringLiteral("shrug"),
+                                               QStringLiteral("fliptable"),
+                                               QStringLiteral("unfliptable"),
+                                               QStringLiteral("sovietflip"),
+                                               QStringLiteral("clear-timeline"),
+                                               QStringLiteral("reset-state"),
+                                               QStringLiteral("rotate-megolm-session"),
+                                               QStringLiteral("md"),
+                                               QStringLiteral("cmark"),
+                                               QStringLiteral("plain"),
+                                               QStringLiteral("rainbow"),
+                                               QStringLiteral("rainbowme"),
+                                               QStringLiteral("notice"),
+                                               QStringLiteral("rainbownotice"),
+                                               QStringLiteral("confetti"),
+                                               QStringLiteral("rainbowconfetti"),
+                                               QStringLiteral("goto"),
+                                               QStringLiteral("converttodm"),
+                                               QStringLiteral("converttoroom")};
+        hasInvalidCommand = !validCommands.contains(commandName);
+    } else
+        hasInvalidCommand = false;
+
+    if (containsInvalidCommand_ != hasInvalidCommand) {
+        containsInvalidCommand_ = hasInvalidCommand;
+        emit containsInvalidCommandChanged();
+    }
+    if (currentCommand_ != commandName) {
+        currentCommand_ = commandName;
+        emit currentCommandChanged();
+    }
```

Comment:
You might want to batch that together. Otherwise the "containsInvalidCommand" property is already true, when the command is still pointing at the old command.

Reply:
What exactly do you mean by batching it together? Emit the signals at the same time? Use one signal for both properties?

Reply:
Yes, emit the signals at the same time. Basically write the changes before you emit the signals. And possibly emit the command changed signal first.